app/models/iiif_canvas_presenter.rb

```diff
@@ -84,12 +88,29 @@ def audio_display_content(quality)
                                            **manifest_attributes(quality, 'Sound'))
     end
 
+    def supplementing_content_data(file)
+      url = if !file.is_a?(SupplementalFile)
+              Rails.application.routes.url_helpers.captions_master_file_url(master_file.id)
+            elsif file.tags.include?('caption')
+              Rails.application.routes.url_helpers.captions_master_file_supplemental_file_url(master_file.id, file.id)
+            elsif file.tags.include?('transcript')
+              Rails.application.routes.url_helpers.transcripts_master_file_supplemental_file_url(master_file.id, file.id)
+            else
+              Rails.application.routes.url_helpers.master_file_supplemental_file_url(master_file.id, file.id)
+            end
+      IIIFManifest::V3::SupplementingContent.new(url, **supplemental_attributes(file))
+    end
+
     def stream_urls
       stream_info[:stream_hls].collect do |d|
         [d[:quality], d[:url]]
       end
     end
 
+    def supplemental_captions_transcripts
+      master_file.supplemental_files(tag: 'caption') + master_file.supplemental_files(tag: 'transcript') + [master_file.captions]
```

Comment:
Do you need to check if `master_file.captions` exists and is persisted?  Otherwise I could image this as adding `nil` or empty captions file to the array which would cause the `captions_master_file_url` be included in the manifest even when it wouldn't return any content from that endpoint.

When I was working on the SpeedyAF PR I found that `master_file.captions` behaves a little differently than I expected:
```
[3] pry(main)> mf.has_captions?
=> false
[4] pry(main)> mf.captions
=> #<IndexedFile uri="http://fedora:8080/fedora/rest/9z/90/2z/84/9z902z84b/captions" >
[5] pry(main)> mf.captions.present?
=> false
[6] pry(main)> mf.captions.persisted?
=> false
[7] pry(main)> mf.captions.content
=> nil
[8] pry(main)> mf.captions.mime_type
=> "text/plain"
```

Reply:
Good catch. Had not noticed that behavior for `master_file.captions` and it was causing `captions_master_file_url` to be erroneously added.